arm/authorization/client.go

```diff
@@ -0,0 +1,759 @@
+}
+
+func (client *ManagementLocksClient) CreateOrUpdateAtResourceGroupLevelRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsPut(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Authorization/locks/{lockName}"))
+}
+
+// CreateOrUpdateAtResourceLevel create or update a management lock at the
+// resource level or any level below resource.;
+//
+// resourceGroupName is the name of the resource group.
+// resourceProviderNamespace is resource identity. parentResourcePath is
+// resource identity. resourceType is resource identity. resourceName is
+// resource identity. lockName is the name of lock. parameters is create or
+// update management lock parameters.
+func (client *ManagementLocksClient) CreateOrUpdateAtResourceLevel(resourceGroupName string, resourceProviderNamespace string, parentResourcePath string, resourceType string, resourceName string, lockName string, parameters ManagementLock) (result ManagementLock, err error) {
+
+	req, err := client.NewCreateOrUpdateAtResourceLevelRequest(resourceGroupName, resourceProviderNamespace, parentResourcePath, resourceType, resourceName, lockName, parameters)
+	if err != nil {
+		return result, fmt.Errorf("authorization: Failure preparing ManagementLocksClient.CreateOrUpdateAtResourceLevel request (%v)", err)
+	}
+
+	req, err = autorest.Prepare(
+		req,
+		client.WithAuthorization(),
+		client.WithInspection())
+	if err != nil {
+		return result, fmt.Errorf("authorization: Failure sending ManagementLocksClient.CreateOrUpdateAtResourceLevel request (%v)", err)
+	}
+
+	resp, err := autorest.SendWithSender(client, req)
+	if err != nil {
+		return result, fmt.Errorf("authorization: Failure sending ManagementLocksClient.CreateOrUpdateAtResourceLevel request (%v)", err)
+	}
+
+	result = ManagementLock{}
+	err = autorest.Respond(
+		resp,
+		client.ByInspecting(),
+		autorest.WithErrorUnlessOK(),
+		autorest.ByUnmarshallingJSON(&result),
+		autorest.ByClosing())
+
+	return result, err
+}
+
+func (client *ManagementLocksClient) NewCreateOrUpdateAtResourceLevelRequest(resourceGroupName string, resourceProviderNamespace string, parentResourcePath string, resourceType string, resourceName string, lockName string, parameters ManagementLock) (*http.Request, error) {
+
+	pathParameters := map[string]interface{}{
+		"lockName":                  url.QueryEscape(lockName),
+		"parentResourcePath":        parentResourcePath,
+		"resourceGroupName":         url.QueryEscape(resourceGroupName),
+		"resourceName":              url.QueryEscape(resourceName),
+		"resourceProviderNamespace": url.QueryEscape(resourceProviderNamespace),
+		"resourceType":              resourceType,
+		"subscriptionId":            url.QueryEscape(client.SubscriptionId),
+	}
+
+	queryParameters := map[string]interface{}{
```

Comment:
@ahmetalpbalkan Cannot be done. The generator does not have sufficient insight. The arguments are declared locally. The "base" lacks knowledge of their presence. Further, in some cases, specific query arguments are necessary which would lead to more complex code in the generator.